CONTRIBUTING.md

```diff
@@ -232,6 +232,36 @@ the event happens.
 Make log message human readable and also include data that is needed for easier
 understanding of what happened and in what context.
 
+### Observability
+
+Out of the box, your users should be able to observe the state of your component. 
+The collector exposes an OpenMetrics endpoint at `/metrics` where your data will land.
+
+When using the regular helpers, you should have some metrics added around key 
+events automatically. For instance, exporters should have `otelcol_exporter_sent_spans` 
+tracked without your exporter doing anything.
+
+In addition to that, go over your code and try to assess what kind of information 
+you'd ask your users if they open a bug report:
+
+- are there internal queues in your components? Add a gauge (`LastValue`) with their states.
+- are you making HTTP calls? Measure the latency and expose a histogram. Consider
+providing a counter with the number of calls that your processor made.
+- are you making sampling decisions? Provide a counter with the outcomes.
+
+At this moment, the collector is still using the OpenCensus Go client for the 
+instrumentation, but that shouldn't stop you from making your component observable.
+As a good practice, have a `metrics.go` with your `stats` and `views`, registering
+the views in `./service/telemetry.go`.
+For the `opentelemetry-collector-contrib` repository, register your view in the
+factory until a more appropriate solution is in place.
```

Comment:
Is this premature to recommend? We have not yet decided how we want to handle this, so encouraging adding more custom metrics is going to make our lives more difficult if we end up deciding to do something that is different from OpenCensus Go metrics.

Reply:
Yes, based on the discussion from last Wednesday, I think we are still defining what's the recommendation. I wrote this before the discussion, under the assumption that this was indeed the recommendation.

Reply:
So, maybe leave out the custom metrics for now and keep the rest?

Reply:
Done